src/actions/postsAction.js

```diff
@@ -0,0 +1,80 @@
+/* @flow */
+import { AsyncStorage } from "react-native";
+import {
+  ADD_POST,
+  DELETE_POST,
+  UPDATE_POST,
+  GET_POSTS
+} from "../constants/actionTypes";
+import testPosts from "../constants/testPosts";
+import type { Post } from "../types/types";
+
+type AddPostAction = { type: "ADD_POST", post: Post };
+type DeletePostAction = { type: "DELETE_POST" };
+type UpdatePostAction = { type: "UPDATE_POST" };
+type GetPostsAction = { type: "GET_POSTS", posts: Array<Post> };
+
+type Action =
+  | AddPostAction
+  | DeletePostAction
+  | UpdatePostAction
+  | GetPostsAction;
+
+type Dispatch = (action: Action | ThunkAction) => any;
+type ThunkAction = (dispatch: Dispatch) => any;
+
+function getIndex(data, id) {
+  let clone = JSON.parse(JSON.stringify(data));
+  return clone.findIndex(obj => parseInt(obj.id) === parseInt(id));
+}
+
+export function addPost(post: Post): ThunkAction {
+  return dispatch => {
+    AsyncStorage.getItem("data", (err, posts) => {
+      posts = JSON.parse(posts);
+      posts.unshift(post);
+      AsyncStorage.setItem("data", JSON.stringify(posts), () => {
+        dispatch({ type: ADD_POST, post: post });
```

Comment:
Use action creators like this one - https://github.com/KseniaKardash/itechart-react-native/pull/10/files#diff-8358153272048e9cdfb62cb76a6dd749R7 , instead of passing objects directly to `dispatch` every time